full-stack-tests/imodelhub-client/src/imodelhub/Briefcases.test.ts

```diff
@@ -149,15 +147,43 @@ describe("iModelHub BriefcaseHandler", () => {
     }
   });
 
-  it("should delete a briefcase (#iModelBank)", async () => {
-    utils.mockGetBriefcase(imodelId, utils.generateBriefcase(2), utils.generateBriefcase(acquiredBriefcaseId));
+  it("should delete a briefcase that does not own locks (#iModelBank)", async () => {
+    let newBriefcase: Briefcase = utils.generateBriefcase(briefcaseId);
+    utils.mockCreateBriefcase(imodelId, undefined, newBriefcase);
+    newBriefcase = await iModelClient.briefcases.create(requestContext, imodelId, newBriefcase);
+
+    utils.mockGetBriefcase(imodelId, newBriefcase);
+    const originalBriefcaseCount = (await iModelClient.briefcases.get(requestContext, imodelId)).length;
+
+    utils.mockGetLocks(imodelId, `?$filter=BriefcaseId+eq+${newBriefcase.briefcaseId}&$top=1`);
+    mockDeleteBriefcase(imodelId, newBriefcase.briefcaseId!);
+    await iModelClient.briefcases.delete(requestContext, imodelId, newBriefcase.briefcaseId!);
+
+    utils.mockGetBriefcase(imodelId);
+    const briefcaseCount = (await iModelClient.briefcases.get(requestContext, imodelId)).length;
```

Comment:
Better check if newBriefcase was deleted. Otherwise this can fail because of some other test created briefcases.

Reply:
Added a check for InstanceNotFound when querying by id